src/tools/Base.js

```diff
@@ -24,7 +24,7 @@ const ToolView = observer(({ item }) => {
 const BaseTool = types
   .model('BaseTool', {
     smart: false,
-    isDrawingTool: false,
+    allowRegionUnselectionOnToolChange: false,
```

Comment:
It's not about letting them to be unselected, that's inevitable behaviour, so I'd shorten name to `unselectRegionOnToolChange` at least (or shorter).
And I think it's good to keep here the most expected value — `true` as usually one tool is for one type of regions, the only grouping we have is Brush+Eraser+MagicWand. And remove this value from usual tools (Rect and others)